nix/shell-plugins.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, config, is-home-manager, ... }:
+with lib;
+let cfg = config.programs._1password-shell-plugins;
```

Comment:
this line lets us reference our own configuration as set by the end user, e.g. it gives us the values the user sets via:

```nix
programs._1password-shell-plugins = {
  # it gives us these values
  enable = true;
  plugins = with pkgs; [ gh awscli2 cachix ];
};
```